pkgs/by-name/cl/clickhouse/generic.nix

```
@@ -0,0 +1,185 @@
+      darwin.bootstrap_cmds
+    ]
+    ++ lib.optionals rustSupport [
+      rustc
+      cargo
+      rustPlatform.cargoSetupHook
+    ];
+
+  buildInputs = lib.optionals stdenv.hostPlatform.isDarwin [ libiconv ];
+
+  dontCargoSetupPostUnpack = true;
+
+  postPatch =
+    ''
+      patchShebangs src/
+      patchShebangs utils/
+
+      sed -i 's|/usr/bin/env perl|"${lib.getExe perl}"|' contrib/openssl-cmake/CMakeLists.txt
+
+      substituteInPlace src/Storages/System/StorageSystemLicenses.sh \
+        --replace-fail '$(git rev-parse --show-toplevel)' "$NIX_BUILD_TOP/$sourceRoot"
+      substituteInPlace utils/list-licenses/list-licenses.sh \
+        --replace-fail '$(git rev-parse --show-toplevel)' "$NIX_BUILD_TOP/$sourceRoot"
+    ''
+    + lib.optionalString stdenv.hostPlatform.isDarwin ''
+      sed -i 's|gfind|find|' cmake/tools.cmake
+      sed -i 's|ggrep|grep|' cmake/tools.cmake
+
+      # Make sure Darwin invokes lld.ld64 not lld.
+      substituteInPlace cmake/tools.cmake \
+        --replace '--ld-path=''${LLD_PATH}' '-fuse-ld=lld'
+    ''
+    + lib.optionalString rustSupport ''
+      cargoSetupPostPatchHook() { true; }
+    '';
+
+  cmakeFlags =
+    [
+      "-DENABLE_CHDIG=OFF"
+      "-DENABLE_TESTS=OFF"
+      "-DENABLE_DELTA_KERNEL_RS=0"
+      "-DCOMPILER_CACHE=disabled"
+    ]
+    ++ lib.optional (
+      stdenv.hostPlatform.isLinux && stdenv.hostPlatform.isAarch64
+    ) "-DNO_ARMV81_OR_HIGHER=1";
+
+  env = {
+    CARGO_HOME = "$PWD/../.cargo/";
+    NIX_CFLAGS_COMPILE =
+      # undefined reference to '__sync_val_compare_and_swap_16'
+      lib.optionalString stdenv.hostPlatform.isx86_64 " -mcx16"
+      +
+        # Silence ``-Wimplicit-const-int-float-conversion` error in MemoryTracker.cpp and
+        # ``-Wno-unneeded-internal-declaration` TreeOptimizer.cpp.
+        lib.optionalString stdenv.hostPlatform.isDarwin
+          " -Wno-implicit-const-int-float-conversion -Wno-unneeded-internal-declaration";
+  };
+
+  # https://github.com/ClickHouse/ClickHouse/issues/49988
+  hardeningDisable = [ "fortify" ];
+
+  postInstall = ''
+    rm -rf $out/share/clickhouse-test
+
+    sed -i -e '\!<log>/var/log/clickhouse-server/clickhouse-server\.log</log>!d' \
+      $out/etc/clickhouse-server/config.xml
+    substituteInPlace $out/etc/clickhouse-server/config.xml \
+      --replace-fail "<errorlog>/var/log/clickhouse-server/clickhouse-server.err.log</errorlog>" "<console>1</console>"
+    substituteInPlace $out/etc/clickhouse-server/config.xml \
+      --replace-fail "<level>trace</level>" "<level>warning</level>"
+  '';
+
+  # Basic smoke test
+  doCheck = true;
+  checkPhase = ''
+    $NIX_BUILD_TOP/$sourceRoot/build/programs/clickhouse local --query 'SELECT 1' | grep 1
+  '';
+
+  # Builds in 7+h with 2 cores, and ~20m with a big-parallel builder.
+  requiredSystemFeatures = [ "big-parallel" ];
+
+  passthru = {
+    tests.clickhouse = if (!lts) then nixosTests.clickhouse else nixosTests.clickhouse-lts;
```

```suggestion
    tests.clickhouse = if lts then nixosTests.clickhouse-lts else nixosTests.clickhouse;
```
Inverting conditions to not have a not usually makes them easier to read and reason about